### PR TITLE
fix dumpEcdsFilterConfigs when not have any ecds configs

### PR DESCRIPTION
**Pull request by zengyuxing007**
Commit Message: Fixes https://github.com/envoyproxy/envoy/issues/32796
Additional Description:   fix dumpEcdsFilterConfigs when never received  any resources
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by adisuissa** [comment]
@yanjunxiang-google IIRC you worked on this previously. Can you please take a first look.
/assign @yanjunxiang-google

**Review by adisuissa — COMMENTED** [review]
Thanks for fixing this issue, overall LGTM.
I may have missed it, but will it be possible to add another test that validates what happens when no ECDS config is received (I only saw this change impacting tests where the config is removed).
Please also add a release note.
/wait

File: test/integration/upstream_network_filter_integration_test.cc [hunk 1]
```diff
@@ -555,14 +555,7 @@ TEST_P(UpstreamNetworkExtensionDiscoveryIntegrationTest, ConfigDumpWithFilterCon
...
   // With /config_dump?resource=ecds_filters, the response has the format: EcdsFilterConfig.
-  envoy::admin::v3::EcdsConfigDump::EcdsFilterConfig ecds_msg;
-  config_dump.configs(0).UnpackTo(&ecds_msg);
-  EXPECT_EQ("", ecds_msg.version_info());
-  envoy::config::core::v3::TypedExtensionConfig filter_config;
-  EXPECT_TRUE(ecds_msg.ecds_filter().UnpackTo(&filter_config));
-  EXPECT_EQ("foo", filter_config.name());
-  // Verify ECDS config dump doesn't have the filter configuration.
-  EXPECT_EQ(false, filter_config.has_typed_config());
+  EXPECT_EQ(0, config_dump.configs_size());
```
[the remark this answers is not in the document]
Comment (zengyuxing007) [reply test/integration/upstream_network_filter_integration_test.cc:558]: Thanks for your review， i'll add some tests and update the comment above.

[the remark this answers is not in the document]
Comment (zengyuxing007) [reply test/integration/upstream_network_filter_integration_test.cc:560]: done

**Comment by zengyuxing007** [comment]
> Can you please also address the comment from above:
> 
> > may have missed it, but will it be possible to add another test that validates what happens when no ECDS config is received (I only saw this change impacting tests where the config is removed).
> > Please also add a release note.

@adisuissa   I think that the existing TTL integration tests have already included tests for when it is unconfigured or not have any configs

File: changelogs/current.yaml [hunk 2]
```diff
@@ -152,6 +152,10 @@ bug_fixes:
   change: |
     When performing a token refresh and forwarding tokens upstream, replace existing token cookies rather than appending as
     another Cookie header.
+- area: xds
```
Comment (adisuissa) [inline changelogs/current.yaml:155]:
```suggestion
- area: admin
```

File: changelogs/current.yaml [hunk 3]
```diff
@@ -152,6 +152,10 @@ bug_fixes:
...
+  change: |
+    Fixed dumpEcdsFilterConfigs when not have any ecds configs.
```
Comment (adisuissa) [inline changelogs/current.yaml:157]:
```suggestion
    Removing the ECDS config entry from the config dump when it does not exist (or has expired).
```

Please double check that what I wrote reflects the actual change.

**Review by adisuissa — COMMENTED** [review]
/assign-from @envoyproxy/senior-maintainers

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!
/assign-from @envoyproxy/senior-maintainers

**Review by zuercher — DISMISSED** [review]

**Comment by zengyuxing007** [comment]
@adisuissa @zuercher @yanjunxiang-google 

Just resolved a merge conflict，PTAL again, thanks!

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Comment by adisuissa** [comment]
Already approved by a senior-maintainer, merging.

**Pull request merged**